src/init.cpp

```diff
@@ -920,6 +921,12 @@ bool AppInitParameterInteraction()
     fCheckBlockIndex = GetBoolArg("-checkblockindex", chainparams.DefaultConsistencyChecks());
     fCheckpointsEnabled = GetBoolArg("-checkpoints", DEFAULT_CHECKPOINTS_ENABLED);
 
+    hashAssumeValid = uint256S(GetArg("-assumevalid", chainparams.GetConsensus().defaultAssumeValid.GetHex()));
```

Comment:
This seems to result in an out-of-bounds memory access when `-assumevalid=` (with no value) is supplied. `uint256S(..)` invokes `base_blob::SetHex(const char *)`, which assumes a supplied length of at least 1. An out-of-bounds access then occurs at [src/uint256.cpp:39](https://github.com/bitcoin/bitcoin/blob/c0ddd32bf629bb48426b0651def497ca1a78e6b1/src/uint256.cpp#L39).